### ISSUE TITLE
Apply file text changes performance problem

### ISSUE THREAD

**Issue by killagu** [dsherret/ts-morph#806]
If the source file is vary large, more than 200,000 lines,  the program will stuck at [getTextFromTextChanges](https://github.com/dsherret/ts-morph/blob/latest/packages/ts-morph/src/manipulation/formatting/getTextFromFormattingEdits.ts#L18).

```js
text = text.slice(0, span.getStart()) + edit.getNewText() + text.slice(span.getEnd());
```

If the text is very long, build a new string cost lot.

We can optimise it with a string array.

```js
const text = sourceFile.getFullText();
let start = 0;
let end = 0;
let edits = [];
for (const { edit } of reversedFormattingEdits) {
    const span = edit.getSpan();
    end = span.getStart();
    const beforeEdit = text.slice(start, end);
    start = end;
    edits.push(beforeEdit);
    edits.push(edit.getNewText());
}
edits.push(text.slice(end));
return edits.join('');
```

If it is ok, I can make a pr to solve the problem.

**Comment by dsherret** [comment]
@killagu for sure! I do this same technique in other places.

**Comment by killagu** [comment]
Would you mind to release a new version with the https://github.com/dsherret/ts-morph/pull/807? I have a project occurred this performance problem.

**Comment by dsherret** [comment]
@killagu published in 7.0.2 🙂
